src/Native/System.Native/pal_errno.cpp

```diff
@@ -299,7 +303,11 @@ extern "C" int32_t SystemNative_ConvertErrorPalToPlatform(Error error)
         case PAL_ENOTEMPTY:
             return ENOTEMPTY;
         case PAL_ENOTRECOVERABLE:
+#if HAVE_ENOTRECOVERABLE
             return ENOTRECOVERABLE;
+#else
+            return -1;
+#endif
```

Comment:
Why return -1 rather just surrounding the whole case in the ifdef and it not match in the switch?